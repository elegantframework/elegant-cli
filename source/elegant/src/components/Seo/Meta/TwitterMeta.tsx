interface Props {
    handle?: string;
    site?: string;
};

/**
 * Twitter card meta data.
 * @returns A list of twitter card meta data items for the seo component.
 */
const TwitterMeta = ({
    handle = "",
    site = "@"
}: Props) => {
    if(handle !== ""){
        // if a site twitter account has been provided
        if(site !== "@"){
            return {
                handle: handle,
                site: site,
                cardType: 'summary_large_image',
            };
        }
        
        // no site provided.. Render only a twitter handle and card
        return {
            handle: handle,
            cardType: 'summary_large_image',   
        };
    }
    else{
        return {
            cardType: 'summary_large_image',   
        };
    }
};

export default TwitterMeta;