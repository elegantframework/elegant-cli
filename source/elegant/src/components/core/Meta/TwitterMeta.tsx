
interface Props {
    /**
     * The website creators Twitter handle. Ex. @tailwindcss
     */
    twitterHandle: string;
};

/**
 * If a twitter handle is provided, return the twitter heading card meta data.
 * @param param0 component props.
 * @returns Twitter site and creator meta heading tags.
 */
const TwitterMeta = ({twitterHandle}: Props) => 
{
    // if there isn't a handle provided, return nothing
    if(twitterHandle === "")
    {
        return null;
    }

    return(
        <>
            <meta key="twitter:site" name="twitter:site" content="@thebrandonowens" />
            <meta key="twitter:creator" name="twitter:creator" content="@thebrandonowens" />
        </> 
    );
}

export default TwitterMeta;