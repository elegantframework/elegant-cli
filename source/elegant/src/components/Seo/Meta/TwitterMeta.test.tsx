import TwitterMeta from "./TwitterMeta";

describe('Twitter meta component', () => {
    it('renders with data properly', () => {
        const result = TwitterMeta({
            handle: "@test"
        });

        expect(result).toMatchSnapshot();
    });

    it('renders with site data properly', () => {
        const result = TwitterMeta({
            handle: "@test2",
            site: "@site_test"
        });

        expect(result).toMatchSnapshot();
    });

    it('renders `the default if nothing is provided`', () => {
        const result = TwitterMeta({
            handle: "",
        });

        expect(result).toMatchSnapshot();
    });
});