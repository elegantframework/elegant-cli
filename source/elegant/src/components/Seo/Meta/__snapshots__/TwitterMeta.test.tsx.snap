// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Twitter meta component renders \`the default if nothing is provided\` 1`] = `
{
  "cardType": "summary_large_image",
}
`;

exports[`Twitter meta component renders with data properly 1`] = `
{
  "cardType": "summary_large_image",
  "handle": "@test",
}
`;

exports[`Twitter meta component renders with site data properly 1`] = `
{
  "cardType": "summary_large_image",
  "handle": "@test2",
  "site": "@site_test",
}
`;
